.github/workflows/sycl_precommit.yml

```diff
@@ -31,19 +31,42 @@ jobs:
     uses: ./.github/workflows/sycl_detect_changes.yml
 
   lint:
-    runs-on: ubuntu-22.04
+    runs-on: cuda
     container:
       image: ghcr.io/intel/llvm/sycl_ubuntu2204_nightly:no-drivers
+      # actions/checkout fails without "--privileged".
+      options: -u 1001:1001 --privileged
     steps:
+    - name: Fake actions/checkout task
+      uses: actions/checkout@v3
+      with:
+        # cached_checkout below uses actions/checkout internally. However, when
+        # actions/checkout is run from within another action step (not from
+        # workflow), github seems to try to download from within the container
+        # and doesn't have requried filesystem permissions. Make sure it's
+        # already downloaded by the time it's needed by checking out some small
```

Comment:
I guess I'm not understanding this. "by checking out some small repository".    Checking out what, exactly?

Reply:
`repository: actions/checkout` - we use it to checkout itself.